UCI-IN4MATX-191-Token-ATM/token-atm-spa
# User Story: Teacher triggers request processing

### Story

As a teacher,
I want Token ATM to have a big button which could be pressed to let Token ATM process all pending requests that have a properly defined and implemented request handler being processed,
so that I can process all pending requests easily.

### Acceptance Criteria

Given that valid Canvas credentials are provided to Token ATM & Token ATM Configuration for the selected Canvas course is valid,
when the teacher clicks the big button for request processing,
then Token ATM will process all pending requests.

### Testing Note

Notes for creating a valid Token ATM Configuration for a Canvas course for testing purposes:

1. Create a new assignment. Record the assignment ID (could be found in the URL when viewing the assignment) as `log_assignment_id`.
2. Create a new quiz. The first question of the quiz shall contain a single option `Request for <token_option_name>`, while `<token_option_name>` is replaced by a name that you want for a token option. Record the quiz ID (could be found in the URL when viewing the quiz) as `quiz_id`.
3. Create a page named `Token ATM Configuration`. Paste the following content with variables replaced in the **Raw HTML Editor** (this is to prevent HTML escape characters from being inserted) (Do not replace the quotation marks around the variables).

    ```html
    <p>
    {
        "log_assignment_id": "<log_assignment_id>",
        "token_option_groups": [
            {
                "name": "Group 1",
                "id": 1,
                "quiz_id": "<quiz_id>",
                "token_options": [
                    {
                        "type": "basic",
                        "id": 1,
                        "name": "<token_option_name>",
                        "token_balance_change": 10
                    }
                ]
            }
        ]
    }
    </p>
    ```
4. Create a user as student to submit the newly created quiz with the only option for the first question selected.
5. Log in to Token ATM and select this course. Press the big button for request processing to trigger it.
6. After the button becomes active again, you should be able to see the grade change and new submission comment from the student's submission page.
7. You could repeat Step 2 to add more token option groups and/or add more token options by adding options to the first question of a quiz. You also need to add corresponding configurations to `token_option_groups` and `token_options` array (shown in Step 3).
